File: TEO/Mav.UI/Mav.Ui/src/app/services/i-question-group.service.ts
import { Observable } from 'rxjs/Observable';
import { QuestionGroupViewModel } from '../models/question-group.viewmodel';
import { AddQuestionGroupViewModel } from '../models/admin/add-question-group.view.model';
import { GenericResult } from '../models/generic.result.model';
import { UpdateQuestionGroupViewModel } from '../models/admin/update-question-group.view.model';

export interface IQuestionGroupService {

    getQuestionGroupById<T>(id: number): Observable<T>;

    getQuestionGroupList(sectionId: number): Observable<QuestionGroupViewModel[]>;

    createQuestionGroup(newItem: AddQuestionGroupViewModel): Observable<GenericResult<number>>;

    updateQuestionGroup(item: UpdateQuestionGroupViewModel): Observable<GenericResult<number>>;

    deleteQuestionGroup(id: number): Observable<GenericResult<number>>;

}

File: TEO/Mav.Services/Abstract/IQuestionGroupService.cs
using Mav.Common.Models;
using Mav.Models.ViewModels;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Mav.Services.Abstract
{
    public interface IQuestionGroupService
    {
        Task<T> GetQuestionGroupById<T>(int id);

        Task<IEnumerable<T>> GetQuestionGroupList<T>(int sectionId);

        Task<GenericResult<T>> CreateQuestionGroup<T>(AddQuestionGroupViewModel m);

        Task<GenericResult<T>> UpdateQuestionGroup<T>(UpdateQuestionGroupViewModel m);

        Task<GenericResult<T>> DeleteQuestionGroup<T>(int id);
    }
}
